azurerm/resource_arm_autoscale_setting.go

```diff
@@ -0,0 +1,887 @@
+									"scale_action": {
+										Type:     schema.TypeList,
+										Required: true,
+										MaxItems: 1,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"direction": {
+													Type:     schema.TypeString,
+													Required: true,
+													ValidateFunc: validation.StringInSlice([]string{
+														string(insights.ScaleDirectionDecrease),
+														string(insights.ScaleDirectionIncrease),
+													}, true),
+													DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+												},
+												"type": {
+													Type:     schema.TypeString,
+													Required: true,
+													ValidateFunc: validation.StringInSlice([]string{
+														string(insights.ChangeCount),
+														string(insights.ExactCount),
+														string(insights.PercentChangeCount),
+													}, true),
+													DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+												},
+												"value": {
+													Type:         schema.TypeInt,
+													Required:     true,
+													ValidateFunc: validation.IntAtLeast(0),
+												},
+												"cooldown": {
+													Type:         schema.TypeString,
+													Required:     true,
+													ValidateFunc: validateIso8601Duration(),
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+						"fixed_date": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"time_zone": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validateAutoScaleSettingsTimeZone(),
+									},
+									"start": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validateRFC3339Date,
+									},
+									"end": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validateRFC3339Date,
+									},
+								},
+							},
+						},
+						"recurrence": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"time_zone": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validateAutoScaleSettingsTimeZone(),
+									},
+									"days": {
+										Type:     schema.TypeList,
+										Required: true,
+										Elem: &schema.Schema{
+											Type: schema.TypeString,
+											ValidateFunc: validation.StringInSlice([]string{
+												"Monday",
+												"Tuesday",
+												"Wednesday",
+												"Thursday",
+												"Friday",
+												"Saturday",
+												"Sunday",
+											}, true),
+											DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+										},
+									},
+									"hours": {
+										Type:     schema.TypeList,
```

Comment:
TODO: Might be better to use Set instead of List here.

Reply:
from the UI (and confirmed via the API) it appears this can only be set to a single value - so there's no need for a Set here:

> The autoscale setting that you provided has an invalid recurrence schedule with invalid count for minutes specified. Specify only one value for minutes.